### PR TITLE
LAB #13

### PR DESCRIPTION
**Pull request by WilfredoN**

### CODE REVIEW

File: lab_13/src/lib.c [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+               current->data.model,
+               current->data.year,
+               current->data.owner.name,
+               current->data.owner.id);
+        current = current->next;
+    }
+    printf("NULL\n");
+}
+
+void display_list_id(forward_list_t *list)
```
Comment (volnsav) [inline lab_13/src/lib.c]: What is the reason to have `display_list1` and `display_list2`?

File: lab_13/src/lib.c [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+{
+    node_t *current = head;
+    while (current != NULL)
+    {
+        printf("Brand: %s, Model: %s, Year: %d, Owner: %s, ID: %s\n",
+               current->data.brand,
+               current->data.model,
+               current->data.year,
+               current->data.owner.name,
+               current->data.owner.id);
```
Comment (volnsav) [inline lab_13/src/lib.c]: Move it in a `display_node`

**Review by volnsav — CHANGES_REQUESTED** [review]
See comments

File: lab_13/src/lib.c (hunk 1 above)
Comment (capynq) [reply lab_13/src/lib.c]: Roudiment from base code, will clean this

File: lab_13/src/lib.c [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+    node_t *current = list->tail;
+    while (current != list->head)
+    {
+        printf("Brand: %s, Model: %s, Year: %d, Owner: %s, ID: %s\n",
+               current->data.brand,
+               current->data.model,
+               current->data.year,
+               current->data.owner.name,
+               current->data.owner.id);
+        current = current->prev;
```
Comment (volnsav) [inline lab_13/src/lib.c]: Use `display_node` instead.

File: lab_13/src/lib.c [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+}
+
+bool is_equal(vehicle_t first, vehicle_t second)
+{
+    return strcmp(first.brand, second.brand) == 0 &&
+           strcmp(first.model, second.model) == 0 &&
+           first.year == second.year &&
+           strcmp(first.owner.name, second.owner.name) == 0 &&
+           strcmp(first.owner.id, second.owner.id) == 0;
+}
```
Comment (volnsav) [inline lab_13/src/lib.c]: `is_node_equal`?

File: lab_13/main.c [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+    vehicle_t data_3;
+    strcpy(data_3.brand, "Ford");
+    strcpy(data_3.model, "Focus");
+    data_3.year = 2019;
+    strcpy(data_3.owner.name, "JD");
+    strcpy(data_3.owner.id, "3");
+    data_3.length = 0;
+    insert_after(list, data_2, data_3);
+
```
Comment (volnsav) [inline lab_13/main.c]: Use an array of structures and initialize all values using the structure syntax.

File: lab_13/include/lib.h [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    struct node_t *next;
+    struct node_t *prev;
+} node_t;
+
+typedef struct forward_list_t
+{
+    node_t *head;
+    node_t *tail;
+} forward_list_t;
+
```
Comment (volnsav) [inline lab_13/include/lib.h]: The `doubly_linked_list`?
The `forward_list`, we expect, has only `next` and `head`

File: lab_13/include/lib.h [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+typedef struct
+{
+    char brand[MAX_NAME_LENGTH];
+    char model[MAX_NAME_LENGTH];
+    int year;
+    owner_t owner;
+    service_t services[MAX_SERVICES_AMOUNT];
+    int length;
+} vehicle_t;
```
Comment (volnsav) [inline lab_13/include/lib.h]: It would be better to have it in a separate header file.

**Review by volnsav — CHANGES_REQUESTED** [review]
Please review and address comments.

File: lab_13/src/lib.c (hunk 3 above)
Comment (capynq) [reply lab_13/src/lib.c]: you mean make separate function to return printf current data?

File: lab_13/src/lib.c [hunk 8]
```diff
@@ -0,0 +1,217 @@
+#include "lib.h"
+
+node_t *create_node(vehicle_t data)
+{
+    node_t *node = (node_t *)malloc(sizeof(node_t));
```
Comment (KononenkoLiza) [inline lab_13/src/lib.c:5]: I think you need to add a memory allocation check.

Comment (capynq) [reply lab_13/src/lib.c:5]: resolved

**Review by volnsav — APPROVED** [review]
Approved

**Pull request merged**